### DISCUSSION TITLE
Cascading Menu & Modal Dialog examples

### DISCUSSION THREAD

**Discussion by jspv** [Textualize/textual#1713] [Show and tell]
Just a few things I've built for my app. 

- **Menu Widget** that supports cascading submenus, mouse and keyboard scrolling with prompts and changing bindings based on the current menu; it also forces focus and removes and restores the original bindings.
- **Yes/No Modal Dialog Widget**  - with customizable prompt and callbacks, also forces focus and removes and restores the original bindings

(side note: this is all developed and running on a Raspberry Pi Zero 2 sitting inside a 3D printed robot - coded and controlled over SSH quite often from an iPad connected via VPN over a mobile hotspot while sitting on a train)

![spotbot](https://user-images.githubusercontent.com/6540751/216213470-61d8ab4f-aead-44cc-8b09-898d751827b5.gif)
![smallmenu](https://user-images.githubusercontent.com/6540751/216213506-a1f4c0e6-6404-445b-90f7-8d7522c2a90e.gif)

Source is here: https://github.com/jspv/spotbot Widgets are in the Widgets folder, sorry no docs yet.  

[edit - apologies, I hadn't pushed the latest code to GitHub, done.  note: the Menu and Dialog widgets are fairly self contained and should be easily reused, you can see how they are called from the main spotbot.py; the actual menus are in add_menus.py]

**Comment by darrenburns** [comment]
Amazing work!

**Reply by jspv** [reply]
Thanks.  Textual is a very well thought through framework, I’m continually impressed with the design (and the performance), and huge props to @davep who didn’t just answer some of my questions, but provided code examples and insight on different approaches.  Next think to tackle will be input widgets.

**Comment by lllama** [comment]
So good.

**Comment by sander76** [comment]
Nice ! I was looking for a dialog window myself. I'll let myself be inspired.
